planning/behavior_velocity_planner/src/scene_module/speed_bump/scene.cpp

```diff
@@ -0,0 +1,187 @@
+
+namespace behavior_velocity_planner
+{
+using motion_utils::calcLongitudinalOffsetPoint;
+using tier4_autoware_utils::createPoint;
+
+namespace
+{
+}  // namespace
+
+SpeedBumpModule::SpeedBumpModule(
+  const int64_t module_id, const int64_t lane_id,
+  const lanelet::autoware::SpeedBump & speed_bump_reg_elem, const PlannerParam & planner_param,
+  const rclcpp::Logger & logger, const rclcpp::Clock::SharedPtr clock)
+: SceneModuleInterface(module_id, logger, clock),
+  module_id_(module_id),
+  lane_id_(lane_id),
+  speed_bump_reg_elem_(std::move(speed_bump_reg_elem)),
+  planner_param_(planner_param),
+  state_(State::SLOW_DOWN),
+  passed_slow_start_point_(false)
+{
+  // point.x : height [m] -- point.y : speed [m/s]
+  Point32 p1;
+  Point32 p2;
+
+  p1.x = 0.05;  // min speed bump height
+  p2.x = 0.30;  // max speed bump height
+
+  p1.y = 2.78;  // [10 kph] max speed for any speed bump
+  p2.y = 1.39;  // [5 kph] min speed for any speed bump
+
+  auto const & constants = getLinearEquation(p1, p2);
+  auto const & m = constants.first;
+  auto const & b = constants.second;
+
+  // Read speed bump height [m] from map
+  speed_bump_height_ =
+    static_cast<float>(speed_bump_reg_elem_.speedBump().attributeOr("height", 0.5));
+
+  // Calculate the speed [m/s] for speed bump
+  speed_bump_slow_down_speed_ = m * speed_bump_height_ + b;
+
+  if (planner_param_.print_debug_info) {
+    std::cout << "------------------------------" << std::endl;
+    std::cout << "Speed Bump ID: " << module_id_ << std::endl;
+    std::cout << "Speed Bump Height [cm]: " << speed_bump_height_ * 100 << std::endl;
+    std::cout << "Calculated Speed [kph]: " << speed_bump_slow_down_speed_ * 3.6 << std::endl;
+    std::cout << "------------------------------" << std::endl;
+  }
+}
+
+bool SpeedBumpModule::modifyPathVelocity(
+  PathWithLaneId * path, [[maybe_unused]] StopReason * stop_reason)
+{
+  debug_data_ = DebugData();
+  debug_data_.base_link2front = planner_data_->vehicle_info_.max_longitudinal_offset_m;
+
+  auto ego_path = *path;
+
+  path_intersects_.clear();
+
+  const auto & ego_pos = planner_data_->current_pose.pose.position;
+  const auto speed_bump = speed_bump_reg_elem_.speedBump();
+
+  const auto intersects =
+    getPolygonIntersects(ego_path, lanelet::utils::to2D(speed_bump).basicPolygon(), ego_pos, 2);
+
+  for (const auto & p : intersects) {
+    path_intersects_.push_back(createPoint(p.x(), p.y(), ego_pos.z));
+  }
+
+  debug_data_.path_polygon_intersection_points = path_intersects_;
+
+  for (const auto & p : speed_bump_reg_elem_.speedBump().basicPolygon()) {
+    debug_data_.speed_bump_polygon.push_back(createPoint(p.x(), p.y(), ego_pos.z));
+  }
+
+  return applySlowDownSpeed(*path);
+}
+
+bool SpeedBumpModule::applySlowDownSpeed(PathWithLaneId & output)
+{
+  if (path_intersects_.empty()) {
+    return false;
+  }
+
+  const auto & ego_pos = planner_data_->current_pose.pose.position;
+  const auto ego_path = output;
```

Comment:
If you don't add reference intentionally, please add a comment about the reason.

Reply:
My mistake, not intentionally, will fix it. Thanks.